feline/src/main/java/com/spotify/feline/AllowancesTransformer.java

```diff
@@ -17,24 +17,44 @@
 
 package com.spotify.feline;
 
-import java.util.Map;
+import java.util.HashSet;
 import java.util.Set;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.ConcurrentMap;
 import net.bytebuddy.agent.builder.AgentBuilder;
 import net.bytebuddy.asm.Advice;
 import net.bytebuddy.description.type.TypeDescription;
 import net.bytebuddy.dynamic.DynamicType;
+import net.bytebuddy.matcher.ElementMatcher;
 import net.bytebuddy.utility.JavaModule;
 
 /**
  * This transformer applies {@link AllowAdvice} to every method registered with {@link
  * Feline#allowBlockingCallsInside(String, String)}.
  */
-class AllowancesTransformer implements AgentBuilder.Transformer {
+class AllowancesTransformer implements AgentBuilder.Transformer, ElementMatcher<TypeDescription> {
 
-  private final Map<String, Set<String>> allowances;
+  private final ConcurrentMap<String, Set<String>> allowances;
 
-  AllowancesTransformer(Map<String, Set<String>> allowances) {
-    this.allowances = allowances;
+  AllowancesTransformer() {
+    this.allowances = new ConcurrentHashMap<>();
+  }
+
+  void allow(final String className, final String methodName) {
+    allowances.compute(
+        className,
+        (key, allowances) -> {
```

Comment:
Nitpick: duplicated names, can we rename the local variable name?

Reply:
done 👍